testing/test_unittest.py

```diff
@@ -735,3 +735,23 @@ def test_skip(self):
         *SKIP*[1]*skipping due to reasons*
         *1 skipped*
     """)
+
+def test_class_method_containing_test_issue1558(testdir):
+    testdir.makepyfile(test_foo="""
+        import unittest
+
+        class MyTestMeta(type):
```

Comment:
I think this test is a little more complex than it needs, although I understand it was your specific use case; IMHO we should keep things simpler in pytest's test suite though, unless we are of course testing for a specific issue with meta-classes and test cases (which is not the actual problem being tested).

I would rather see something like:

```python
class MyTestCase(unittest.TestCase):
    
    def test_should_run(self):
        pass
    def test_should_not_run(self):
        pass

MyTestCase.test_should_not_run.__test__ = False
```

What do you think?